mycroft/audio/services/mpv/__init__.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2017 Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from mycroft.audio.services import AudioBackend
+from mycroft.util.log import LOG
+try:
+    from mpv import MPV
+except ImportError:
+    LOG.error("install mpv with "
+              "pip install git+https://github.com/JarbasAl/python-mpv/")
+    raise
+
+
+class MPVService(AudioBackend):
+    def __init__(self, config, emitter=None, name='mpv'):
+        super(MPVService, self).__init__(config, emitter)
+        self.player = MPV(ytdl=True)
+        self.player._set_property('video', 'no')
+        self.name = name
+        self.tracks = []
+        self.index = 0
+
+    def track_start(self, data, other):
+        if self._track_start_callback:
+            self._track_start_callback(self.track_info()['name'])
+
+    def supported_uris(self):
+        return ['file', 'http', 'https']
+
+    def clear_list(self):
+        self.tracks = []
+
+    def add_list(self, tracks):
+        self.tracks += tracks
+        LOG.info("Track list is " + str(self.tracks))
+
+    def play(self):
+        LOG.info('MpvService Play')
+        if len(self.tracks):
+            self.player.stop()
+            self.player.play(self.tracks[self.index])
+
+    def stop(self):
+        LOG.info('MpvService Stop')
+        self.clear_list()
+        self.player.command("stop")
+
+    def pause(self):
+        LOG.info('MpvService Pause')
+        self.player._set_property("pause", True)
```

Comment:
the class does not expose pause functionality, this method must be used, however i can change that if we require my forked repo, original repo does not work in python2

Reply:
Briefly reading the module's source (so this might very well be wrong) it looks like the self.player._set_property(a,b) could be done via self.player['a'] = b. Seem to add some extra stuff to the parameters setting the property'd context so I'm not 100% sure it'll be exactly equivalent.